pkg/lib/bundle/interpreter.go

```diff
@@ -0,0 +1,54 @@
+package bundle
+
+import (
+	"fmt"
+	"path"
+	"sort"
+
+	"github.com/operator-framework/operator-registry/pkg/registry"
+)
+
+type bundleDirInterpreter struct {
+	bundleCsvName string
+	pkg           *registry.Package
+}
+
+func NewBundleDirInterperter(bundleDir string) (*bundleDirInterpreter, error) {
+	csv, err := registry.ReadCSVFromBundleDirectory(bundleDir)
+	if err != nil {
+		return nil, fmt.Errorf("error loading CSV from bundle directory, %v", err)
+	}
+
+	pkgDir, err := registry.NewPackageGraphLoaderFromDir(path.Join(bundleDir, ".."))
```

Comment:
I'm not sure this is what we want to do here.

This reads in the directory of files and generates a graph, and tells you if there are problems.

But the generate command is used for a single bundle. That single bundle may be applied to any target index, and has nothing to do with the bundles that happen to be in in the filesystem next to it.

I think this would be fine without the graph loader in this PR at all. Thoughts?

Reply:
But I thought the whole purpose of this though is to automagically decide what channels a given bundle is. How can we do that without attempting to build the graph for a given package?

Reply:
Same ask!

Reply:
Thanks for the explanation - commented before ☕️ !

I think there's a slight impedance mismatch here - this option is now only really useful if you're attempting convert an entire directory of the old format, right?

Should this really be part of the normal command for generating/buliding single bundles when it's so tied to the set?

Reply:
I think the trigger for it should be "if you didn't pass in these parameters explicitly, let's go see if we can find them for you" rather than it being an explicit opt in/opt out feature. IMO that's a better experience

Reply:
I second that this should be an experience enhancement type of feature based on the fact that you have the full-blown format with bundles nested in a package along with other bundles. Without it, it does not have any information we would be looking for. The `default channel` and `channels` are only obtainable in the `*.package.yaml` file. To figure out which channels a bundle belongs to require relationships of this bundle with respect to every channel heads and see if it is directly/transitively replaced by those channel heads to be included as part of that channel. Without the package and rest of the bundles, all we can be certain from a single bundle is the `package name`.

After all, this feature is just auto-filling `package name`, `channels`, and `default channels` for a bundle.